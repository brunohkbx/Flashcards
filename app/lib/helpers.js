import { Platform } from 'react-native';

export const waitFor = ms => {
  return new Promise(resolve => setTimeout(resolve, ms));
};

export const getOS = () => {
  return Platform.OS;
};
