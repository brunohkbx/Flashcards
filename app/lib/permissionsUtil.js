import { Permissions } from 'expo';

export const obtainNotificationPermission = async () => {
  const { status: existingStatus } = await Permissions.getAsync(
    Permissions.NOTIFICATIONS
  );
  let finalStatus = existingStatus;

  if (existingStatus !== 'granted') {
    const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);

    finalStatus = status;
  }

  return finalStatus;
};
