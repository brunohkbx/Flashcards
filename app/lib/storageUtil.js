import { AsyncStorage } from 'react-native';
import { pickBy, assign } from 'lodash/object';

const DECK_STORAGE_KEY = 'flashcards:deck';

export const getDecks = () => {
  return AsyncStorage.getItem(DECK_STORAGE_KEY);
}

export const createDeck = deck => {
  return AsyncStorage.mergeItem(DECK_STORAGE_KEY, JSON.stringify(deck));
}

export const deleteDeck = id => {
  return getDecks()
    .then(decks => JSON.parse(decks))
    .then(decks => pickBy(decks, deck => deck.id !== id))
    .then(decks => AsyncStorage.setItem(DECK_STORAGE_KEY, JSON.stringify(decks)));
}

export const editDeck = deck => {
  return getDecks()
    .then(decks => JSON.parse(decks))
    .then(decks => assign(decks, deck))
    .then(decks => AsyncStorage.setItem(
      DECK_STORAGE_KEY, JSON.stringify(decks)
    ));
}
