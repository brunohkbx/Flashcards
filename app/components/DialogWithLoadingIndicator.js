import React from 'react';
import PropTypes from 'prop-types';
import { ActivityIndicator, View } from 'react-native';
import { Paragraph, Portal, Dialog } from 'react-native-paper';
import { getOS } from '../lib/helpers';

const DialogWithLoadingIndicator = ({ visible, loadingMessage }) => {
  return (
    <Portal>
      <Dialog visible={visible} dismissable={false} >
        <Dialog.Content>
          <View style={{flexDirection: 'row', alignItems: 'center'}}>
            <ActivityIndicator
              size={getOS() === 'ios' ? 'large' : 48}
              style={{ marginRight: 16 }}
            />
            <Paragraph>{loadingMessage}</Paragraph>
          </View>
        </Dialog.Content>
      </Dialog>
    </Portal>
  );
};

DialogWithLoadingIndicator.propTypes = {
  visible: PropTypes.bool,
  loadingMessage: PropTypes.string
};

DialogWithLoadingIndicator.defaultProps = {
  visible: false,
  loadingMessage: ''
};

export default DialogWithLoadingIndicator;
