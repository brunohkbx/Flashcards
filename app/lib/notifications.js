import { Notifications } from 'expo';
import * as storageUtil from '../lib/storageUtil';

export const clearLocalNotification = () => {
  return storageUtil.unscheduleNotification()
    .then(Notifications.cancelAllScheduledNotificationsAsync);
};

export const scheduleLocalNotification = () => {
  return storageUtil.isScheduledNotification()
    .then(JSON.parse)
    .then(data => {
      if (data === null) {
        Notifications.cancelAllScheduledNotificationsAsync();

        let tomorrow = new Date();
        tomorrow.setDate(tomorrow.getDate() + 1);
        tomorrow.setHours(18);
        tomorrow.setMinutes(30);

        Notifications.scheduleLocalNotificationAsync(
          {
            title: 'Haven\'t seen you in a while',
            body: 'ðŸ‘‹ Don\'t forget to take your quiz today!',
            ios: {
              sound: true
            },
            android: {
              sound: true,
              priority: 'high',
              sticky: false,
              vibrate: true
            }
          },
          {
            time: tomorrow,
            repeat: 'day',
          }
        );

        storageUtil.scheduleNotification();
      }
    });
};
