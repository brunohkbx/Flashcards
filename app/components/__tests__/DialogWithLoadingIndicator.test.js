import React from 'react';
import { shallow } from 'enzyme';
import DialogWithLoadingIndicator from '../DialogWithLoadingIndicator';
import * as helpers from '../../lib/helpers';

describe('DialogWithLoadingIndicator', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      visible: true,
      loadingMessage: 'Editing.....'
    }, propOverrides);

    const wrapper = shallow(<DialogWithLoadingIndicator {...defaultProps} />);

    return { wrapper };
  };

  it('renders properly on android', () => {
    const { wrapper } =  setup();
    helpers.getOS = jest.fn(() => 'android');

    expect(wrapper).toMatchSnapshot();
  });

  it('renders properly on ios', () => {
    const { wrapper } =  setup();
    helpers.getOS = jest.fn(() => 'ios');

    expect(wrapper).toMatchSnapshot();
  });
});
