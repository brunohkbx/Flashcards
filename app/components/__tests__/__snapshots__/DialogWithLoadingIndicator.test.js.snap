// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogWithLoadingIndicator renders properly on android 1`] = `
<withTheme(Portal)>
  <withTheme(Dialog)
    dismissable={false}
    visible={true}
  >
    <Dialog.Content>
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "marginRight": 16,
            }
          }
        />
        <Paragraph>
          Editing.....
        </Paragraph>
      </Component>
    </Dialog.Content>
  </withTheme(Dialog)>
</withTheme(Portal)>
`;

exports[`DialogWithLoadingIndicator renders properly on ios 1`] = `
<withTheme(Portal)>
  <withTheme(Dialog)
    dismissable={false}
    visible={true}
  >
    <Dialog.Content>
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size={48}
          style={
            Object {
              "marginRight": 16,
            }
          }
        />
        <Paragraph>
          Editing.....
        </Paragraph>
      </Component>
    </Dialog.Content>
  </withTheme(Dialog)>
</withTheme(Portal)>
`;
